### PR TITLE
Update EIP-5749: Move to last call

### PR DESCRIPTION
**Pull request by kvhnuke**
please refer to https://github.com/ethereum/EIPs/pull/5912#issuecomment-1314247015 to comments regarding title and the reason for `evmproviders`

### CODE REVIEW

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Line 18 has an incomplete sentence.
Line 33 has a random `..etc`. This should be removed, since it doesn't add anything and doesn't make sense in the context of the list ("et cetera" is not an example).
Line 35 has a small mistake: it says "only display" when it should really be "display" (web3modal and web3onboard will still show potentially uninstalled options by the nature that they don't inject either `window.ethereum` or `window.evmproviders`)
Line 68: The "well" in "well documented" can be removed
Line 71 and 78: ```` ``` ```` should be replaced with ```` ```text````
Line 87-91: This list should be inlined so that it's all one sentence.
Line 122: "make sure" should be replaced with "ensure"

**Comment by kvhnuke** [comment]
@Pandapip1 let me know if I miss anything, all requested changes should be done

File: EIPS/eip-5749.md [hunk 1]
```diff
@@ -64,17 +65,17 @@ The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
...
+Type `EIP1193Provider` is documented at [EIP-1193](./eip-1193.md)
 
-```
+```text
 interface ProviderInfo
 uuid: UUIDv4 unique to the provider
 name: Name of the Wallet
 icon: base64 encoded svg image
 description: Description for your wallet
 ```
```
Comment (Pandapip1) [inline EIPS/eip-5749.md:76]: What you have is non-standard notation. I'd recommend using typedoc, which is standardized. In addition, the properties are underspecified:
```suggestion
```typescript
/**
 * Represents the assets needed to display a wallet
 */
interface ProviderInfo {
   /**
    * A UUIDv4 unique to the wallet provider.
    *
    * This must remain the same across versions but must be different across channels. For example, MetaMask, MetaMask Flask, and Trust Wallet should each have different UUIDs, but MetaMask 10.22.2 and MetaMask 9.8.1 should have the same UUID.
    *
    * @readonly
    */
    uuid: string;
    /**
     * The name of the wallet provider (e.g. `MetaMask` or `Trust Wallet`)
     *
     * @readonly
     */
    name: string;
    /**
     * A base64 encoded SVG image.
     *
     * Base64 is defined in RFC 4648.
     *
     * @readonly
     */
    icon: string;
    /**
     * A description of the wallet provider.
     *
     * @readonly
     */
    description: string;
}
```
```

File: EIPS/eip-5749.md [hunk 2]
```diff
@@ -64,17 +65,17 @@ The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
...
 
-```
+```text
 interface EVMProvidersType
 key is RECOMMENDED to be the name of the extension
 ```
```
Comment (Pandapip1) [inline EIPS/eip-5749.md:81]:
```suggestion
```typescript
/**
 * The type of `window.evmproviders`
 */
interface EVMProviders {
  /**
   * The key is RECOMMENDED to be the name of the extension in snake_case. It MUST contain only lowercase letters, numbers, and underscores.
   */
  [index: string]: string;
}
```
```

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: EIPS/eip-5749.md [hunk 3]
```diff
@@ -83,11 +84,7 @@ By adopting an object for EIP-1193 compliant providers we can have multiple diff
...
 ## Rationale
 
-By introducing `ProviderInfo` type web onboarding libraries such as
-	Web3Modal
-	Web3React
-	Web3Onboard
-can easily grab the necessary information to populate their popup window to choose the wallet.
+By introducing `ProviderInfo` type web onboarding libraries such as Web3Modal, Web3React, Web3Onboard can easily grab the necessary information to populate their popup window to choose the wallet.
 
 The name `evmproviders` was chosen in order to be inclusive of other evm-compliant chains.
```
[the remark this answers is not in the document]
Comment (kvhnuke) [reply EIPS/eip-5749.md:123]: yea I was trying to make a point on why SVG images are better, Im rewriting this sentence

[the remark this answers is not in the document]
Comment (kvhnuke) [reply EIPS/eip-5749.md:123]: @Pandapip1 does the new sentence makes more sense?

File: EIPS/eip-5749.md [hunk 4]
```diff
@@ -29,75 +30,87 @@ At present, `window.ethereum` is the prevailing method by which Ethereum-compati
...
-	Web3React
-	Web3Onboard
-can easily grab the necessary information to populate their popup window to choose the wallet.
+Standardizing a `ProviderInfo` type allows determining the necessary information to populate a wallet selection popup. This is particularly useful for web3 onboarding libraries such as Web3Modal, Web3React, and Web3Onboard.
 
-The name `evmproviders` was chosen in order to be inclusive of other evm-compliant chains.
+The name `evmproviders` was chosen to include other EVM-compliant chains.
 
-**data:image/svg+xml;** svg data uri was chosen since it is easier to be modified if the application requires for example different size for the image.
+**data:image/svg+xml;** SVG image type was chosen in order to have more flexibility in modifying images, such as resizing.
```
Comment (Pandapip1) [inline EIPS/eip-5749.md:108]:
```suggestion
The SVG image format was chosen for its flexibility, lightweight nature, and dynamic resizing capabilities.
```
How about this?

**Review by Pandapip1 — COMMENTED** [review]
Apart from this one minor nit, LGTM

File: EIPS/eip-5749.md [hunk 5]
```diff
@@ -4,7 +4,8 @@ title: The 'window.evmproviders' object
 description: Add 'window.evmproviders' and suggest the eventual removal of 'window.ethereum'
 author: Kosala Hemachandra (@kvhnuke), Brett Kolodny (@brettkolodny)
 discussions-to: https://ethereum-magicians.org/t/eip-5749-deprecate-window-ethereum/11195
-status: Review
+status: Last Call
+last-call-deadline: 2022-12-05
```
Comment (Pandapip1) [inline EIPS/eip-5749.md:8]:
```suggestion
last-call-deadline: 2022-12-07
```

File: EIPS/eip-5749.md (hunk 4 above)
Comment (kvhnuke) [reply EIPS/eip-5749.md:108]: perfect, thank you!

**Review by kdenhartog — CHANGES_REQUESTED** [review]
How do you expect this to interact with EIP-5593? I'd think we should resolve this before moving this to last call otherwise we may hit incompatibilities that have to be resolved separately. Out of curiosity how many wallets have reviewed this work currently and are supportive of it? Also are you aware that MetaMask is looking to start modifying their ETH provider based on Chain Agnostic Standards Alliance's CAIP-25 so much of this work may become incompatible with the largest wallet in this space?

For me, I don't object to the content in this, but I do to the quick status update without making sure that wallet providers plan to implement this breaking change and that this work doesn't interfere with other efforts to update the provider object.

**Comment by kvhnuke** [comment]
> How do you expect this to interact with EIP-5593? I'd think we should resolve this before moving this to last call otherwise we may hit incompatibilities that have to be resolved separately. Out of curiosity how many wallets have reviewed this work currently and are supportive of it? Also are you aware that MetaMask is looking to start modifying their ETH provider based on Chain Agnostic Standards Alliance's CAIP-25 so much of this work may become incompatible with the largest wallet in this space?
> 
> For me, I don't object to the content in this, but I do to the quick status update without making sure that wallet providers plan to implement this breaking change and that this work doesn't interfere with other efforts to update the provider object.

I believe discussions regarding the proposal should happen under the discussion link, but to answer your question, 
this will note interfere with EIP-5593, it is still upto the extension wallet to decide when to inject and it could be based on user settings.

Whole goal of this EIP is for wallets to open up a new avenue so dapps can detect which wallets are installed by the user, fight 
 for `window.ethereum` will continue to happen for the near future as this EIP is not a breaking change. However, this EIP will let wallets to inject also under `window.evmproviders={}` 
we already received support from few major wallets and major web3 onboarding libraries. 

chain agnostic standards are different topic where we have to redo the whole communication layer, I think we should visit that once that kind of standard turn into a reality

**Comment by Pandapip1** [comment]
I'd say it would be up to EIP-5593 to specify when `window.evmproviders` is injected. Compatibility with EIP-5593 seems out of the scope of this EIP, but compatibility with this EIP seems in the scope of EIP-5593.

**Review by Pandapip1 — APPROVED** [review]
LGTM.

**Comment by kvhnuke** [comment]
> LGTM.

Thank you!!!

**Pull request merged**